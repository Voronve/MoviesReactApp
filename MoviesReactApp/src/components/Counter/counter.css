.counterContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.counterNumber {
    margin-top: 5px;
    font-size: 2rem;
    margin-bottom: 1rem;
    padding: 0.5rem;
    background-color: #f0f7ff;
    border: 1px solid #d4ebff;
    border-radius: 4px;
}

.buttonContainer {
    display: flex;
    flex-direction: row;
    align-items: center;
}

button {
    font-size: 1rem;
    cursor: pointer;
    padding: 0.5rem 1rem;
    margin: 0.5rem;
    border: none;
    border-radius: 4px;
    color: white;
    background-color: #2196f3;
    box-shadow: 1px 1px 3px rgba(0, 0, 0, 0.25);
    transition: background-color 0.3s;
    min-width: 200px;
}

button:active {
    box-shadow: inset 1px 1px 3px rgba(0, 0, 0, 0.25);
}

button.increment:hover {
    background-color: #42a5f5;
}

button.decrement:hover {
    background-color: #1e88e5;
}