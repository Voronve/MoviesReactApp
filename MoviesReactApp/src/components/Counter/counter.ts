import React, { Component } from 'react';
import './counter.css';

 /**
 * Default value when user opens the page
 */
interface counterProps {
    /**
     * Default value when user opens the page
     */
    initialValue: number
};
 /**
 * Default value when user opens the page
 */
interface CounterState {
    value: number
};

/**
 * Default value when user opens the page
 */
export class Counter extends Component<counterProps> {
    state = {} as CounterState;

    constructor(props: {initialValue: number}) {
        super(props);
        this.state = {
          value: props.initialValue ?? 0,
        };
    }

    handleIncrement = () => {
        this.setState( (prevState: CounterState) => {
            return { value: ++prevState.value }
        });
    }

    handleDecrement = () => {
        this.setState( (prevState: CounterState) => {
            return { value: (prevState.value > 0 ? --prevState.value  : prevState.value) }
        });
    }

    render() {
        return(
            React.createElement(
                "div",
                { className: "counterContainer " },
                React.createElement(
                    "div",
                    {
                        className: "counterNumber",
                        children: this.state.value
                    }),
                React.createElement(
                    "div",
                    {
                        className: "buttonContainer",
                    },
                    React.createElement(
                        "button",
                        {
                            className: "increment",
                            onClick: this.handleIncrement
                        },
                        "Push to add one"),
                    React.createElement(
                        "button",
                        {
                            className: "decrement",
                            onClick: this.handleDecrement
                        },
                        "Push to substract one")
                ),
            )
        )
    }
}