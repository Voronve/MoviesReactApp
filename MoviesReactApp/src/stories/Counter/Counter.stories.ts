import type { Meta, StoryObj } from '@storybook/react';
import { Counter } from '../../components/Counter/counter';

const meta = {
    title: 'Old components/Counter',
    component: Counter,
    parameters: {
      layout: 'centered',
    },
    tags: ['autodocs'],
    argTypes: {
      initialValue: { control: 'number' },
    },
  } satisfies Meta<typeof Counter>;

  export default meta;
  type Story = StoryObj<typeof meta>;

  // More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
  export const Initial: Story = {
    args: {
      initialValue: 5
    },
  };