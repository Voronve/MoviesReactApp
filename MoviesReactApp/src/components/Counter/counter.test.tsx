import Counter from './counter';
import { fireEvent, render, screen } from '@testing-library/react';

describe("Testing Counter component", () => {

    test("Testing Counter snapshot", () => {
        const { asFragment } = render(<Counter initialValue={0}/>);

        expect(asFragment()).toMatchSnapshot();
    });

    test("Testing Counter initial value", () => {
        const initialValue = 554;

        render(<Counter initialValue={initialValue}/>);

        expect(screen.getByText('554')).toBeInTheDocument();
    });

    test("Testing Counter value after increment", () => {
        const initialValue = 0;

        render(<Counter initialValue={initialValue}/>);

        fireEvent.click(screen.getByText("Push to add one"));
        expect(screen.queryByText('0')).toBeNull();
        expect(screen.getByText('1')).toBeInTheDocument();
    });

    test("Testing Counter value after decrement (shouldn't be lower then zero)", () => {
        const initialValue = 1;

        render(<Counter initialValue={initialValue}/>);
        const decrementButton = screen.getByText("Push to substract one");
        fireEvent.click(decrementButton);
        fireEvent.click(decrementButton);
        expect(screen.getByText('0')).toBeInTheDocument();
    });
});