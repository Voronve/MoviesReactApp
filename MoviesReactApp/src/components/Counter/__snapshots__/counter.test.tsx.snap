// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Counter component Testing Counter snapshot 1`] = `
<DocumentFragment>
  <div
    class="counterContainer "
  >
    <div
      class="counterNumber"
    >
      0
    </div>
    <div
      class="buttonContainer"
    >
      <button
        class="increment"
      >
        Push to add one
      </button>
      <button
        class="decrement"
      >
        Push to substract one
      </button>
    </div>
  </div>
</DocumentFragment>
`;
